### DISCUSSION TITLE
GameObjectEvent with non-prefab GameObject?

### DISCUSSION THREAD

**Discussion by ArnaudBBRI** [unity-atoms/unity-atoms#401] [Q&A]
Hello,

I observe (via the editor) that GameObjectEvents require a **prefab** GameObject as an argument, which I understand makes sense since ScriptableObjects are data on disk, and hence can't be "temporary" objects that only live in a Scene.

Is it possible to somehow pass a non-prefab GameObject via these events? If not, what would be the best approach? I guess using an IntEvent and passing the InstanceID would be one, is it good practice? 

Thanks a lot!
Arnaud

**Comment by soraphis** [comment]
when invoking the event at runtime you can send any runtime object you want.

**Reply by ArnaudBBRI** [reply]
Oh ok, great then! 
Thanks a lot!!
